### PR TITLE
WIP: init graphql support

### PR DESCRIPTION
**Pull request by a1012112796**
not like #12285 use [gqlgen](https://github.com/99designs/gqlgen) as frame

add two operations as init
query: viewer
mutation: createRepository

### CODE REVIEW

**Comment by SirMishaa** [comment]
Up ? Do you need some help for this PR ?

I'm waiting a lot the support for GraphQL.
It is game changer.

File: Makefile [hunk 1]
```diff
@@ -273,6 +274,12 @@ errcheck:
 	@echo "Running errcheck..."
 	@errcheck $(GO_PACKAGES)
 
+.PHONY: generate-graphql
+generate-graphql:
+	cd modules/graphql && \
+	mod=vendor $(GO) run github.com/99designs/gqlgen genrate && \
```
Comment (lafriks) [inline Makefile:280]: Shouldn't there be `generate`?

File: go.mod [hunk 2]
```diff
@@ -29,7 +30,7 @@ require (
 	github.com/ethantkoenig/rupture v1.0.0
 	github.com/gliderlabs/ssh v0.3.1
 	github.com/glycerine/go-unsnap-stream v0.0.0-20190901134440-81cf024a9e0a // indirect
-	github.com/go-chi/chi v1.5.1
+	github.com/go-chi/chi v3.3.2+incompatible
```
Comment (lafriks) [inline go.mod:33]: This seems incorrect, it would be ok to upgrade to v4

**Comment by wxiaoguang** [comment]
Is it feasible for current Gitea code base?

**Pull request closed without merging**